lib/parslet/atoms/entity.rb

```diff
@@ -22,9 +23,11 @@ def try(source, context, consume_all)
   end
   
   def parslet
-    @parslet ||= @block.call.tap { |p| 
-      raise_not_implemented unless p
-    }
+    return @parslet unless @parslet.nil?
+    @parslet = @block.call
+    raise_not_implemented if @parslet.nil?
+    @parslet.label = @label
```

Comment:
I don't understand why the label needs to be copied along the hierarchy here. Disabling this leads to no test failures. Why are you coping labels around here? Can we do without this change? Can you add some tests here?